src/main/java/org/agrona/concurrent/ringbuffer/OneToOneRingBuffer.java

```diff
@@ -85,7 +85,8 @@ public boolean write(final int msgTypeId, final DirectBuffer srcBuffer, final in
 
         final AtomicBuffer buffer = this.buffer;
         final int recordLength = length + HEADER_LENGTH;
-        final int requiredCapacity = align(recordLength, ALIGNMENT);
+        final int alignedRecordLength = align(recordLength, ALIGNMENT);
+        final int requiredCapacity = alignedRecordLength + HEADER_LENGTH;
```

Comment:
Why do we require the extra capacity?
Before: requiredCapacity = align(length + HEADER_LENGTH, ALIGNMENT)
After:    requiredCapacity = align(length + HEADER_LENGTH, ALIGNMENT) + HEADER_LENGTH